pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java

```diff
@@ -136,41 +135,76 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
     @VisibleForTesting
     Collection<PulsarSplit> getSplitsPartitionedTopic(int numSplits, TopicName topicName, PulsarTableHandle
             tableHandle, SchemaInfo schemaInfo, TupleDomain<ColumnHandle> tupleDomain) throws Exception {
+
         int numPartitions;
         try {
-            numPartitions = (this.pulsarAdmin.topics().getPartitionedTopicMetadata(topicName.toString())).partitions;
+            numPartitions = (getPulsarAdmin().topics().getPartitionedTopicMetadata(topicName.toString())).partitions;
         } catch (PulsarAdminException e) {
             if (e.getStatusCode() == 401) {
                 throw new PrestoException(QUERY_REJECTED,
-                        String.format("Failed to get metadata for partitioned topic %s: Unauthorized", topicName));
+                    String.format("Failed to get metadata for partitioned topic %s: Unauthorized", topicName));
             }
 
             throw new RuntimeException("Failed to get metadata for partitioned topic "
-                    + topicName + ": " + ExceptionUtils.getRootCause(e).getLocalizedMessage(),e);
+                + topicName + ": " + ExceptionUtils.getRootCause(e).getLocalizedMessage(),e);
         }
 
-        int actualNumSplits = Math.max(numPartitions, numSplits);
+        List<Integer> predicatePartitions = new ArrayList<>();
```

Comment:
Move line 152-181 to a function "getPredicatedPartitions" or "getFilteredPartitions"

Reply:
ok